### PR TITLE
Experimenting with emotion integration

### PR DESCRIPTION
**Pull request by dprokop**
Wasn't that obvious to enable. Had to modify webpack's config a little bit: 
- enabled `ts-loader` in `webpack.hot.js`
- enabled `babel-loader` in `webpack.dev.js`
- enabled `babel-plugin-emotion` in both of these ^^

In `@grafana/ui/DeleteButton` there is small sample of both css-in-js and styled approach from emotion. Note, that for emotion to work with TS we need to annotate files using `css` prop (https://emotion.sh/docs/css-prop) with JSX pragma. This is sth that I don't necessarily like. I haven't succeeded with `@emotion/babel-preset-css-prop` preset though...

I'd love to hear your opinion. Also, i'd like to take a look at `styled-components` and `JSS` before making any decisions here.

### CODE REVIEW

File: packages/grafana-ui/src/components/DeleteButton/DeleteButton.tsx [hunk 1]
```diff
@@ -32,34 +43,74 @@ export class DeleteButton extends PureComponent<Props, State> {
...
     return (
-      <span className="delete-button-container">
-        <a className={'delete-button ' + showDeleteButton + ' btn btn-danger btn-small'} onClick={this.onClickDelete}>
+      <span css={styles.container}>
+        <a
+          css={css`
+            ${styles.deleteButton};
+            ${!this.state.showConfirm ? styles.deleteButtonVisible : ''};
+          `}
+          className="btn btn-danger btn-small"
```
Comment (dprokop) [inline packages/grafana-ui/src/components/DeleteButton/DeleteButton.tsx]: `css` / `className` composition enables us to use both "legacy" and newly created styles

**Comment by dprokop** [comment]
For styled-components POC see https://github.com/grafana/grafana/commit/c70fb5a68c91afce32b3fb7e66674aa96689e58f

Integration went way easier. Just installed the package and was ready to use. But the `styled.X` approach does not resonate with me as well as `css` prop from emotion.

**Comment by torkelo** [comment]
Can we close this now that the buttons PR was merged?

**Comment by dprokop** [comment]
Closing, Emotion integrated with https://github.com/grafana/grafana/pull/16228

**Pull request closed without merging**